cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go

```diff
@@ -48,12 +48,16 @@ func Convert_v1beta3_InitConfiguration_To_kubeadm_InitConfiguration(in *InitConf
 
 // Convert_kubeadm_JoinConfiguration_To_v1beta3_JoinConfiguration converts a private JoinConfiguration to public JoinConfiguration.
 func Convert_kubeadm_JoinConfiguration_To_v1beta3_JoinConfiguration(in *kubeadm.JoinConfiguration, out *JoinConfiguration, s conversion.Scope) error {
+	// Migrate the discovery timeout.
+	out.Discovery.Timeout = in.Timeouts.Discovery.DeepCopy()
```

Comment:
can in.Timeouts.Discovery be nil?

Reply:
oh, `(*Duration) DeepCopy()` is nil-safe, nevermind

Reply:
looks like `in.Timeouts` can be nil though?

Reply:
i think only in unit tests and non-user facing code paths, because of this line which is there to make the fuzzer happy:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go#L62

Reply:
ok, I'm not familiar with how these are constructed, didn't know the risk there